tests/js/core/widget/core/Spin/Spin.js

```diff
@@ -1,181 +1,352 @@
+					loop: "disabled",
+					labels: ["Aaa", "Bbb", "Ccc", "Ddd"],
+					digits: 3, // 0 - doesn't complete by zeros
+					dragTarget: "self"
+				},
+				spinWidget = tau.engine.instanceWidget(spin, "core.Spin", options);
+
+			equal(spinWidget.option("min"), options.min, "Option min of Spin is " + options.min);
+			equal(spinWidget.option("max"), options.max, "Option max of Spin is " + options.max);
+			equal(spinWidget.option("moduloValue"), options.moduloValue,
+				"Option moduloValue of Spin is " + options.moduloValue);
+			equal(spinWidget.option("shortPath"), options.shortPath,
+				"Option shortPath of Spin is " + options.shortPath);
+			equal(spinWidget.option("duration"), options.duration,
+				"Option duration of Spin is " + options.duration);
+			equal(spinWidget.option("rollHeight"), options.rollHeight,
+				"Option rollHeight of Spin is " + options.rollHeight);
+			equal(spinWidget.option("itemHeight"), options.itemHeight,
+				"Option itemHeight of Spin is " + options.itemHeight);
+			equal(spinWidget.option("momentumLevel"), options.momentumLevel,
+				"Option momentumLevel of Spin is " + options.momentumLevel);
+			equal(spinWidget.option("scaleFactor"), options.scaleFactor,
+				"Option scaleFactor of Spin is " + options.scaleFactor);
+			equal(spinWidget.option("moveFactor"), options.moveFactor,
+				"Option moveFactor of Spin is " + options.moveFactor);
+			equal(spinWidget.option("loop"), options.loop,
+				"Option loop of Spin is " + options.loop);
+			equal(spinWidget.option("labels"), options.labels,
+				"Option labels of Spin is " + options.labels);
+			equal(spinWidget.option("digits"), options.digits,
+				"Option digits of Spin is " + options.digits);
+			equal(spinWidget.option("dragTarget"), options.dragTarget,
+				"Option dragTarget of Spin is " + options.dragTarget);
+
+			spinWidget.destroy();
+		});
+
+		test("Undefined values of options Spin test", 4, function () {
+			var spin = document.getElementById("spin"),
+				options = {
+					min: undefined,
+					max: undefined,
+					duration: undefined
+				},
+				spinWidget = tau.engine.instanceWidget(spin, "core.Spin", options);
+
+			spinWidget.value(undefined);
+
+			equal(spinWidget.option("min"), 0, "Spin option min with undefined value check");
+			equal(spinWidget.option("max"), 0, "Spin option max with undefined value check");
+			equal(spinWidget.option("duration"), 0, "Spin option duration with undefined value check");
+			spinWidget.value(0);
+			equal(spinWidget.value(), 0, "Default option value of Spin is " + 0);
+
+			spinWidget.destroy();
+		});
+
+		test("Values Spin test", 4, function () {
+			var spin = document.getElementById("spin"),
+				options = {
+					min: 3,
+					max: 10,
+					loop: "disabled"
+				},
+				testValues = [0, 10, 100, 11],
+				spinWidget = tau.engine.instanceWidget(spin, "core.Spin", options);
+
+			spinWidget.value(testValues[0]);
+			equal(spinWidget.value(), testValues[0], "Spin value check (0)");
```

Comment:
I think we expected the "3" because "3" is the min value.
Why is the value "0"?

Reply:
This part of tests I moved from mobile profile. I'm not sure... I have to check this.

Reply:
I know where is the issue. When loop is disabled and and new value is out of scope the widget is ignoring new value.
I'll change this behaviour to limit to min / max.
```
		prototype._setValue = function (value) {
			var self = this;

			value = window.parseFloat(value, 10);
			// @todo: for spin with labels the textContent should contains label by value;
			self._ui.placeholder.textContent = value;

			if (isNaN(value)) {
				ns.warn("Spin: value is not a number");
			} else if (value !== self.options.value) {
				if (value >= self.options.min && value <= self.options.max || self.options.loop === "enabled") {

					self._previousCount = self._count;
```

Reply:
Fixed.